website/src/docs/migration-guides.md

```diff
@@ -8,7 +8,117 @@ category: "Docs"
+  * reason: transition to ESM
+  * migrate: import the `Uppy` class by default and/or use named exports for everything else.
+
+#### `@uppy/aws-s3-multipart`
+
+* Remove `client` getter and setter.
+  * reason: internal usage only
+  * migrate: use exposed options only
+
+#### `@uppy/react`
+
+* Don’t expose `validProps` on the exported components.
+  * reason: internal usage only
+  * migrate: use exposed options only
+
+#### `@uppy/store-redux`
+
+* Remove backwards compatible exports (static properties on `ReduxStore`). Exports, such as `reducer`, used to also be accesible on the `ReduxStore` export. This has now been removed.
+  * reason: transition to ESM
+  * migrate: use named imports
+
+#### `@uppy/thumbnail-generator`
+
+* Remove `rotateImage`, `protect`, and `canvasToBlob` from plugin prototype.
+  * reason: internal usage only
+  * migrate: use exposed options only
+
+## Migrate from Companion 3.x to 4.x
+
+### Minimum required Node.js version is v14.20.0
+
+Aligning with the Node.js [Long Term Support (LTS) schedule](https://nodejs.org/en/about/releases/) and to use modern syntax features.
+
+### `companion.app()` returns `{ app, emitter }` instead of `app`
+
+Companion 3.x provides the emitter as `companionEmitter` on `app`. As of 4.x, an object is returned
+with an `app` property (express middleware) and an `emitter` property (event emitter). This
+provides more flexibility in the future and follows best practices.
+
+### Removed compatibility for legacy Custom Provider implementations
+
+[Custom Provider](https://uppy.io/docs/companion/#Adding-custom-providers)
+implementations must use the Promise API. The callback API is no longer supported.
+
+### Default to no ACL for AWS S3
+
+Default to no [ACL](https://docs.aws.amazon.com/AmazonS3/latest/userguide/acl-overview.html) for S3 uploads. Before the default was `public-read` but AWS now discourages ACLs. The environment variable `COMPANION_AWS_DISABLE_ACL` is also removed, instead Companion only uses `COMPANION_AWS_ACL`.
+
+### `protocol` sent from Uppy in any `get` request is now required (before it would default to Multipart).
+
+If you use any official Uppy plugins, then no migration is needed. For custom plugins that talk to Companion, make to send along the `protocol` header with a value of `multipart`, `s3Multipart`, or `tus`.
+
+### `emitSuccess` and `emitError` are now private methods on the `Uploader` class.
+
+It’s unlikely you’re using this, but it’s technically a breaking change.
+In general, don’t depend on implicitly internal methods, use exposed APIs instead.
+
+### Removed `chunkSize` backwards compatibility for AWS S3 Multipart
+
+`chunkSize` option will now be used as `partSize` in AWS multipart. Before only valid values would be respected. Invalid values would be ignored. Now any value will be passed on to the AWS SDK, possibly throwing an error on invalid values.
+
+### Removed backwards compatibility for `/metrics` endpoint
+
+TODO @mifi
```

Comment:
@mifi please make a suggestion here what this should be

Reply:
Metrics will now always be served under `options.server.path`. Before 4.x, it would be served under the root.

Reply:
Can you make a code suggestion with a short explanation what metrics are and this comment?

Reply:
The `metrics` option is a boolean flag to tell Companion whether to provide an endpoint `/metrics` with Prometheus metrics. Metrics will now always be served under `options.server.path`. Before v4.x, it would always be served under the root.

E.g. if `{ options: { metrics: true, server: { path: '/companion' } } }`, metrics will now be served under `/companion/metrics`. In v3.x, the metrics would be served under `/metrics` in this case.